geniza/annotations/migrations/0005_annotation_cleanup_nbsp.py

```diff
@@ -0,0 +1,27 @@
+# Generated by Django 3.2.16 on 2023-10-20 18:51
+
+from django.db import migrations
+
+
+def cleanup_annotations(apps, schema_editor):
+    """Cleanup all annotations with unicode nbsp in their labels or values"""
+    Annotation = apps.get_model("annotations", "Annotation")
+    for anno in Annotation.objects.filter(content__icontains="\xa0"):
+        for body in anno.content["body"]:
+            if "label" in body:
+                body["label"] = body["label"].replace("\xa0", " ")
+            if "value" in body:
+                body["value"] = body["value"].replace("\xa0", " ")
```

Comment:
Do you need to normalize spaces here or anywhere else you're cleaning up these non-breaking spaces?  i.e., if there are non-breaking spaces and regular white spaces together, you'll end up with extra spaces. In most contexts I think it won't matter, since in both Solr and typical browser html display multiple spaces get displayed as a single space. The places I'm thinking of that matter are when we display transcriptions (we have a css style to preserve white space) and the corpus export.

Since it's a limited set you're cleaning up, and you've already looked at them, maybe you already know this isn't a concern.

Reply:
Thank you so much for catching this, it was something I completely overlooked, and in fact it did occur over a hundred times in annotations. Not sure about document descriptions but I'm sure it was present there as well. Used regex to improve the replacement in all cases!